### PR TITLE
Bandchain.js: sendtx

### LINKED ISSUE #2888: Implement `send_tx` (block, sync, async)

### PR DESCRIPTION
**Pull request by Pitchanai**
Fixed: #2888 

## Implementation details
- Send tx

## Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [x] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [x] The pull request includes a description of the implementation/work done in detail.
- [x] The pull request includes any and all appropriate unit/integration tests
- [ ] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [x] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

File: helpers/bandchain2.js/__tests__/client/sendTx.test.ts [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+      sequence: '1092',
+    },
+  ],
+}
+
+const mockTxHash =
+  'E204AAD58ACA8F00942B1BB66D9F745F5E2C21E04C5DF6A0CB73DF02B6B51121'
+
+let client = new Client(TEST_RPC)
+// let client = new Client(MASTER)
```
Comment (evilpeach) [inline helpers/bandchain2.js/__tests__/client/sendTx.test.ts]: remove

File: helpers/bandchain2.js/src/client.ts [hunk 2]
```diff
@@ -1,20 +1,39 @@
...
-  private async get(path: string, param?: object) {
-    const response = await axios.get(`${this.rpcUrl}${path}`, param)
+  private async get(path: string, options?: object) {
+    const response = await axios.get(`${this.rpcUrl}${path}`, options)
     return response.data
   }
 
-  private async getResult(path: string, param?: object) {
-    const response = await this.get(`${path}`, param)
+  private async post(path: string, param: object) {
```
Comment (pzshine) [inline helpers/bandchain2.js/src/client.ts:25]: Can you change `param` -> `option`?

File: helpers/bandchain2.js/src/client.ts [hunk 3]
```diff
@@ -1,20 +1,39 @@
...
+    const response = await axios.post(`${this.rpcUrl}${path}`, param)
```
Comment (pzshine) [inline helpers/bandchain2.js/src/client.ts:26]: Here as well

File: helpers/bandchain2.js/src/client.ts [hunk 4]
```diff
@@ -1,20 +1,39 @@
...
+    return response.data
+  }
+
+  private async getResult(path: string, options?: object) {
+    const response = await this.get(`${path}`, options)
+    return response.result
+  }
+
+  private async postResult(path: string, param: object) {
+    const response = await this.post(`${path}`, param)
```
Comment (pzshine) [inline helpers/bandchain2.js/src/client.ts:36]: please change here too

File: helpers/bandchain2.js/__tests__/client/sendTx.test.ts [hunk 5]
```diff
@@ -0,0 +1,353 @@
+// import { jsxText } from '@babel/types'
```
Comment (evilpeach) [inline helpers/bandchain2.js/__tests__/client/sendTx.test.ts]: remove?

**Review by pzshine — CHANGES_REQUESTED** [review]

File: helpers/bandchain2.js/src/client.ts [hunk 6]
```diff
@@ -1,20 +1,39 @@
...
+    const response = await axios.post(`${this.rpcUrl}${path}`, param)
+    return response.data
+  }
+
+  private async getResult(path: string, options?: object) {
+    const response = await this.get(`${path}`, options)
+    return response.result
+  }
+
+  private async postResult(path: string, param: object) {
```
Comment (evilpeach) [inline helpers/bandchain2.js/src/client.ts]: data

File: helpers/bandchain2.js/src/client.ts (hunk 4 above)
Comment (evilpeach) [inline helpers/bandchain2.js/src/client.ts]: Let's use data instead of param

**Review by evilpeach — CHANGES_REQUESTED** [review]

File: helpers/bandchain2.js/src/data.ts [hunk 7]
```diff
@@ -14,9 +17,74 @@ export class Coin {
...
+
+export interface ReferenceDataUpdated {
+  base: number
+  quote: number
+}
+
+export interface ReferenceData {
+  pair: string
+  rate: number
+  updated: ReferenceDataUpdated
```
Comment (evilpeach) [inline helpers/bandchain2.js/src/data.ts]: updatedAt

**Review by evilpeach — CHANGES_REQUESTED** [review]

**Comment by evilpeach** [comment]
@Pitchanai Please resolve the conflict

**Review by pzshine — APPROVED** [review]

File: helpers/bandchain2.js/src/client.ts [hunk 8]
```diff
@@ -66,6 +132,157 @@ export default class Client {
...
+  //   askCount: number,
+  // ): Promise<RequestInfo> {
+  //   const response = await this.getResult(`/oracle/request_search`, {
+  //     oid: oid,
+  //     calldata: calldata,
+  //     min_count: minCount,
+  //     ask_count: askCount,
+  //   })
+  //   return response
+  // }
```
Comment (evilpeach) [inline helpers/bandchain2.js/src/client.ts]: ???

**Review by evilpeach — CHANGES_REQUESTED** [review]

**Review by evilpeach — APPROVED** [review]

**Pull request merged**